packages/tests/server/minitrpc.ts

```diff
@@ -0,0 +1,98 @@
+import type {
+  AnyMutationProcedure,
+  AnyProcedure,
+  AnyQueryProcedure,
+  AnyRouter,
+  ProcedureArgs,
+  ProcedureRouterRecord,
+} from '@trpc/server';
+import type { TRPCResponse } from '@trpc/server/rpc';
+import type { inferTransformedProcedureOutput } from '@trpc/server/shared';
+
+interface ProxyCallbackOptions {
+  path: string[];
+  args: unknown[];
+}
+
+type ProxyCallback = (opts: ProxyCallbackOptions) => unknown;
+
+const noop = () => {
+  // noop
+};
```

Comment:
This can actually be inlined.

I've forgotten the reason why this needs to be specifically a `Function` and not like `null` or something, might be worth pointing out